index.bs

```diff
@@ -199,23 +196,30 @@ Abstract Operations {#abstract-operations}
 
 Automation {#automation}
 ==========
-This section extends the [=automation=] section defined in the Generic Sensor API [[GENERIC-SENSOR]]
-to provide mocking information about the proximity level for the purposes of
-testing a user agent's implementation of {{ProximitySensor}} API.
 
-<h3 id="mock-proximity-sensor-type">Mock Sensor Type</h3>
+This section extends [[GENERIC-SENSOR#automation]] by providing [=Proximity Sensor=]-specific virtual sensor metadata.
 
-The {{ProximitySensor}} class has an associated [=mock sensor type=] which is
-<a for="MockSensorType" enum-value>"proximity"</a>, its [=mock sensor reading values=]
-dictionary is defined as follows:
+The [=per-type virtual sensor metadata=] [=map=] must have the following [=map/entry=]:
+: [=map/key=]
+:: "`proximity`"
+: [=map/value=]
+:: A [=virtual sensor metadata=] whose [=virtual sensor metadata/virtual sensor type=] is [=Proximity Sensor=] and [=reading parsing algorithm=] is [=proximity reading parsing algorithm=].
 
-<pre class="idl">
-  dictionary ProximityReadingValues {
-    required double? distance;
-    required double? max;
-    required boolean? near;
-  };
-</pre>
+<h3 dfn>Proximity reading parsing algorithm</h3>
+<div algorithm="proximity reading parsing algorithm">
+    : input
+    :: |parameters|, a JSON {{Object}}
+    : output
+    :: A [=sensor reading=] or **undefined**
+
+    1.  Let |reading| be a new [=sensor reading=].
+    1.  Let |value| be a new [=sensor reading=].
+    1. Let |key| be the [=/list=] « "`distance`", "`max`", "`near`" ».
+    1.  [=map/For each=] |key| → let |value| be the result of [=parse single-value number reading=] with |parameters| and |key|.
```

Comment:
- This is not the right format. `key` is a list, not a map, see https://infra.spec.whatwg.org/#list-iterate.
- `|value|` should be declared in a separate substep.
- [`near`](https://w3c.github.io/proximity/#dom-proximitysensor-near) is a boolean, so you cannot use "parse single-value number reading" to parse it. You'll need some custom steps to parse this value. For example:

```diff
diff --git a/index.bs b/index.bs
index 4c7a0a5..c887f88 100644
--- a/index.bs
+++ b/index.bs
@@ -43,6 +43,9 @@ urlPrefix: https://w3c.github.io/sensors; spec: GENERIC-SENSOR
     text: fingerprinting; url: device-fingerprinting
     text: user identifying; url: user-identifying
     text: mitigation strategies; url: mitigation-strategies
+urlPrefix: https://w3c.github.io/webdriver/; spec: WEBDRIVER2
+  type: dfn
+    text: get a property; url: dfn-getting-properties
 </pre>
 <pre class="link-defaults">
 spec:infra;
```

and then

```
- Let |near| be the result of invoking [=get a property=] from |parameters| with "near".
- If |near|'s type is not Boolean, return **undefined**.
- Set reading["near"] to |near|.
```

Reply:
Hopefully fixed in https://github.com/w3c/proximity/pull/57/commits/28ddd12525270a98cd6f42f37ec1345a14eb9994